Doc/library/typing.rst

```diff
@@ -1500,6 +1509,93 @@ These are not used in annotations. They are building blocks for declaring types.
    ``True`` as the value of the ``total`` argument. ``True`` is the default,
    and makes all items defined in the class body required.
 
+   It is possible for a ``TypedDict`` type to inherit from one or more other ``TypedDict`` types
+   using the class-based syntax.
+   Usage::
+
+      class Point3D(Point2D):
+          z: int
+
+   ``Point3D`` has three items: ``x``, ``y`` and ``z``. It is equivalent to this
+   definition::
+
+      class Point3D(TypedDict):
+          x: int
+          y: int
+          z: int
+
+   A ``TypedDict`` cannot inherit from a non-TypedDict class,
+   notably including :class:`Generic`. For example::
+
+      class X(TypedDict):
+          x: int
+
+      class Y(TypedDict):
+          y: int
+
+      class Z(object): pass  # A non-TypedDict class
+
+      class XY(X, Y): pass  # OK
+
+      class XZ(X, Z): pass  # raises TypeError
+
+      T = TypeVar('T')
+      class XT(X, Generic[T]): pass  # raises TypeError
+
+   The type info for introspection can be accessed via ``Point2D.__annotations__``,
```

Comment:
Maybe we can link the annotations howto here for `__annotations__` instead of asking the user to directly access the field.

Reply:
I agree this!
 Also, `__annotations__` is not a best practice in Python versions 3.10 and newer, perhaps it is not appropriate to be emphasized here. 

I will delete the section of `.. attribute:: __annotations__`, and suggest user to access `annotations-howto` directly.